.github/workflows/docs.yml

```diff
@@ -0,0 +1,58 @@
+name: API Reference
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+    tags:
+      - 'v*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      deployments: write
+      id-token: write
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
+
+      - name: Set up Go 1.19
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.19
+
+      - name: Download Packages
+        run: |
+          go install golang.org/x/tools/cmd/godoc@v0.3.0
+          go install github.com/johnstarich/go/gopages@v0.1.16
+
+#      - uses: ably/sdk-upload-action@v1
+#        id: sdk-upload-prempt
+#        with:
+#          mode: preempt
+#          artifactName: godoc
+
+      - name: Build Documentation
+        run: >
+          gopages
+          -source-link "https://github.com/ably/ably-go/blob/${{ github.sha }}/ably/{{slice .Path 5}}{{if .Line}}#L{{.Line}}{{end}}"
+          -brand-description "Go client library for Ably realtime messaging service."
+          -brand-title "Ably Go SDK"
+          -base ./../../../../..
+#          -base $BASE_URL
```

Comment:
It would be nice to remove this commented out line, given there's no accompanying commentary explaining why it's been kept here.

Reply:
> Was the issue with base URL sorted out? We're yet to complete [ably/sdk-upload-action#32](https://github.com/ably/sdk-upload-action/issues/32) 🤔

It's temporarily sorted by specifying the baseUrl. For permanent fix, we need to resolve the issue.

Reply:
removed, but we need to update this once we fix https://github.com/ably/sdk-upload-action/issues/32